### DISCUSSION TITLE
not-found.js page is continuously reloading.

### DISCUSSION THREAD

**Discussion by ajay9758** [vercel/next.js#51958] [App Router] [answered]
i m using next js 13.4 app router but my custom not-found.js page is continuously reloading till 404 page is visible.But the default next js 404 is not reloading continuously. is there any thing i m missing out.

**Comment by vignesh-gupta** [comment]
Could you provide some more info on it? and also your page code.

I'll suggest check if you are accidentally calling the component recursively. Or check for state change in any parent element.

**Reply by ajay9758** [reply]
'use client'
import NextLink from "next/link";
import { Box, Button, Container, SvgIcon, Typography } from "@mui/material";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";

const Page = () => (
  <>
    <Box
      component="main"
      sx={{
        alignItems: "center",
        display: "flex",
        flexGrow: 1,
        minHeight: "100%",
      }}
    >
      <Container maxWidth="md">
        <Box
          sx={{
            alignItems: "center",
            display: "flex",
            flexDirection: "column",
          }}
        >
          <Box
            sx={{
              mb: 3,
              textAlign: "center",
            }}
          >
            <img
              alt="Under development"
              src="/assets/images/error-404.png"
              style={{
                display: "inline-block",
                maxWidth: "100%",
                width: 400,
              }}
            />
          </Box>
          <Typography align="center" sx={{ mb: 3 }} variant="h3">
            404: The page you are looking for isn’t here
          </Typography>
          <Typography align="center" color="text.secondary" variant="body1">
            You either tried some shady route or you came here by mistake.
            Whichever it is, try using the navigation
          </Typography>
          <Button
            component={NextLink}
            href="/"
            startIcon={
              <SvgIcon fontSize="small">
                <ArrowBackIcon />
              </SvgIcon>
            }
            sx={{ mt: 3 }}
            variant="contained"
          >
            Go back to dashboard
          </Button>
        </Box>
      </Container>
    </Box>
  </>
);

export default Page;

**Comment by icyJoseph** [comment, marked as answer]
This is a bit of "expected" strange behaviour.

This happens on Dev Mode only, when the 404 is triggered on purpose, and not through calls to `notFound` (I think) - and it is the way the Next.js server looks for pages that maybe didn't exist, now, but you maybe create on your text editor a few moments later.

You can think of it doing polling to check if the page is still, not existing. 

I know it might be annoying to properly do development, because it keeps, reloading and you miss out on state or what not...

**Reply by ajay9758** [reply]
Thank you sir for your response , reloading only happening in dev mode you are right.
